roles/security/tasks/main.yml

```diff
@@ -0,0 +1,103 @@
+---
+- name: Update APT packages cache
+  apt: update_cache={{ update_apt_cache }} cache_valid_time=86400
+
+- name: Perform aptitude safe-upgrade
+  apt: upgrade=yes
+  when: run_aptitude_safe_upgrade is defined and run_aptitude_safe_upgrade
+
+- name: add user
+  user: name="{{ server_user }}" shell="{{ shell }}" password="{{ server_user_password }}"
+  when: server_user != "root"
+
+- name: Add authorized_keys for the user
+  authorized_key: user={{ server_user }} key="{{ lookup('file', item) }}"
+  with_items:
+    - "{{ user_public_keys }}"
+
+- name: install sudo
+  apt: name=sudo update_cache={{ update_apt_cache }} state=present cache_valid_time=86400
+
+- name: add user to sudoers
+  lineinfile: dest=/etc/sudoers
+              regexp="{{ server_user }} ALL"
+              line="{{ server_user }} ALL=(ALL) ALL"
+              state=present
+  when: server_user != "root"
+
+# Unattended Upgrades settings
```

Comment:
Just to keep things a bit more organized, can we put these security features into separate yml files.  For example, `setup_unattended_upgrades.yml`.  Then in `main.yml` we can do an include with a `when:` condition for that include.  This way we'll also only need to use the `when:` condition once per security feature.